session/session.go

```diff
@@ -77,6 +95,7 @@ func NewZKSession(servers string, recvTimeout time.Duration, logger stdLogger) (
 		events:        events,
 		subscriptions: make([]chan<- ZKSessionEvent, 0),
 		log:           logger,
+		quit: 				 quit,
```

Comment:
Indentation is off here - you could also just do the `make(chan string)` right here instead of declaring it above. I think you can do this more idiomatically by declaring this as a `chan bool`, and then closing the channel in the test instead.